### ISSUE TITLE
Komodo Keeps Crashing

### ISSUE THREAD

**Issue by mjbrancato** [Komodo/KomodoEdit#3189]
### Short Summary
Komodo IDE 11 still crashes constantly just like Komodo 10: Email: mike@mjbcode.com.
Log
  File "/home/mjbrancato/Komodo-IDE-11/lib/python/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
gaierror: [Errno -2] Name or service not known

### Steps to Reproduce
Use Komodo IDE 10 or 11 for more than 30 minutes

### Expected results
Application does not crash

### Actual results
Application constantly crashes

### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?* 11
*Operating System (and version)?* Xubuntu 16.0.4.2 LTS

**Comment by Naatan** [comment]
Thanks for the report @mjbrancato. The crash reports unfortunately don't tell me much other than it is likely a threading issue.

Are you using remote files or version control by any chance? Those would be the two most likely culprits.

**Comment by DEklof** [comment]
I am having the same issue.  I have no real use for the new features of Komodo 11, in fact some, like the symbol tables (that works only in Safe mode  on my installation I consider to be inferior to Komodo 10 version.  I purchased V11 anyway in hopes  that the crashing problem had been fixed.  It has not.  The constant crashes and restarts put a serious dent in my producivity.

**Comment by Naatan** [comment]
Please let's stay on-topic. I'm sorry to hear you don't like the upgrade, let's try and fix the issues that are not by-design.

Same question for you: Are you using remote files or version control by any chance?

**Comment by mjbrancato** [comment]
I do work with remote files from time to time, but I also work out of the GIT path as well using the Komodo Project Publishing Tools. I don't use Komodo's version control because there is no GitFlow support and projects are harder to manage so I use SmartGit.

Is there anything else I can provide? It usually happens when I run it for awhile or sometimes when waking up from sleep. The early versions of 9 and 10 didn't have this issue.

I also need to figure out how to get notified on your responses as well.

**Comment by Naatan** [comment]
You may not use Komodo's version control but it is still active. Can you try disabling it under Preferences > Source Code Control > Git (checkbox next to the name). See if that stops the crashes?

**Comment by mjbrancato** [comment]
I did like that you implemented a version control feature, but yes, I never touched it or used it so that isn't a problem to disable. I disabled GIT and I guess Bazaar existed too so I disabled that too. Only thing I can do is let you know if there are no crashes.

Thank you.

**Comment by Naatan** [comment]
Be sure to also let me know if there are NO crashes. That'd tell me the issue is with version control.

**Comment by mjbrancato** [comment]
I will, I didn't have any crashes today before disabling version control, but I wasn't actively working on any files (Usually, I keep Komodo open along with all my other apps during a work week).

Let me use it for a good active week, and if there are no crashes, then we can assume it's version control. I will investigate some things on my end as well and report my findings.

Was Bazaar a feature of Komodo 9? And how about a feature of the very early versions of Komodo 10 and/or enabled by default if installed on system?

**Comment by Naatan** [comment]
Bazaar has been in Komodo for as long as the other VCS systems have. I think it was introduced back in version 5 or around that time.

**Comment by mjbrancato** [comment]
I don't recall it crashing on my laptop and Bazaar option is greyed out because it's not on that Laptop.

My desktop is sometimes a guinea pig since I make a lot of modifications and then replicate the working ones to the laptop. So some package I might have been using was dependent on Bazaar. I am going to enable GIT and keep Bazaar disabled.

I will report my findings in a few days. If it still crashes with Bazaar disabled, I will disable GIT too. This is a quicker way for me to test it, since we think that disabling Version Control is the issue. Meaning if it was GIT issue, then it should crash with GIT enabled.

**Comment by Naatan** [comment]
I'm confused, why do you think Bazaar is related? You said you use GIT.

**Comment by mjbrancato** [comment]
Bazaar is enabled on my desktop and not on my laptop (Bazaar shouldn't even be on my computers, I never knew it existed). I don't recall my laptop crashing? And it was recent with the later versions of Komodo 10 and on.

GIT is enabled on both systems, but I do not use the GIT feature in Komodo at all. You do know your software better though since you guys developed it. I just have a strange reverse troubleshooting process. Like what if I disable and the next crash is in a week or it doesn't crash anymore and we say it is a GIT issue, but what if it's not? They happen randomly, sometimes every 45 minutes, sometimes they take awhile. Or it depends on how much I am interacting. I will collect all my findings and report back to you.

**Comment by Naatan** [comment]
Whether you use it is not really relevant, if the feature is enabled and Komodo sees your files use this VCS then it will activate that functionality. I would suggest you try disabling Git, from what you've described I don't see Bazaar being an issue as it doesn't sound like you actually use Bazaar.

**Comment by mjbrancato** [comment]
I will do that as well. Wasn't trying to argue, sorry, just trying to quickly troubleshoot. Definitely boosting development productivity with all these new features and fixes!

Komodo 11 is not bad at all, despite if it appears like I have a lack of appreciation. Just a few issues like everything else on first launch, but a couple minor versions will make it the best. :-)

**Comment by Naatan** [comment]
> I will do that as well. Wasn't trying to argue, sorry, just trying to quickly troubleshoot. 

Oh sorry, I didn't mean to come across as argumentative. I tend to just be a bit to the point .. :p

**Comment by mjbrancato** [comment]
No, I didn't mean that, sorry. I just wanted to let you know that I wasn't. People usually see me as an argumentative person if I don't do what they say immediately with no side comments.

**Comment by Naatan** [comment]
As long as you remain constructive with your feedback I don't mind at all :)

**Comment by mjbrancato** [comment]
@Naatan so I disabled GIT and Bazaar and after about 5 to 6 hours of using it, it crashed again.

It was right when I pressed Ctrl + Home it just crashed.

**Comment by Naatan** [comment]
Ok, next could you try disabling Preferences > Editor > Confirmation Dialogs > Detect when files are changed outside the environment

**Comment by codermjb** [comment]
I disabled that feature and restarted Komodo. I will let you know.

**Comment by codermjb** [comment]
I got another crash even with the Version Control and detect when files are changed are disabled.

**Comment by Naatan** [comment]
How much time went between those crashes?

**Comment by codermjb** [comment]
Usually about an hour, but seems less lately. But, I haven't been working on much code this week so it's hard to tell. I will have to get back to you. It's usually when I am actively working on Komodo.

**Comment by Naatan** [comment]
Can you try disabling Preferences > Editor > Confirmation Dialogs > Detect when files are changed outside the environment.

**Comment by codermjb** [comment]
@Naatan I had already disabled that 7 days ago :-) http://prntscr.com/gw5dp8

**Comment by Naatan** [comment]
Hmm alright. Do you regularly use debugging? What's your general workflow when you're actively using Komodo?

**Comment by codermjb** [comment]
I don't use debugging tools although that would be futuristic for me (nice to use if I could).

General WorkFlow task:
- I pull up the website I am working on in Google Chrome.
- I review and confirm what needs to be done/changed/fixed.
- I open Komodo (usually already open on a business week)
- I open up the project (contains local folder, remote folder, and usually setup for Publishing)
- I open up the file from local path, modify it, save it (auto upload is enabled).
- If it doesn't push it, I do a force push for the file or whatever I need to to get the file uploaded.
- Go back to Google Chrome, refresh page and verify changes are complete
- Then I repeat this.

Once changes are good, I use SmartGit with a feature branch (I use GitFlow) to commit my work, then finish feature. Then push changes to GIT.

**Comment by Naatan** [comment]
I see. The publishing would be a suspect here, does the crash happen to correlate to a push by chance?

**Comment by codermjb** [comment]
I never got a crash with Publishing/Saving. I may sometimes have failed attempts at publishing, but I found a work around that I mentioned in the other issue thread.

I want to say it's when I am mainly scrolling, or one time I pressed Ctrl + Home to get to the top of the page and it crashed? Or sometimes when I switch to another file to reference too while I am working on another one. I feel it does it more when I am heavily using it (nothing crazy, maybe 5-8 tabs open).

**Comment by DEklof** [comment]
I have been following this dialog and have tried all of the fixes suggested.  Komodo still crashes frequently. I have been working using debugging for the last 4 hours and have had to restart at least 5 times.  The problem is even worse than it was in V10.

**Comment by Naatan** [comment]
Does it happen when you aren't debugging?

**Comment by DEklof** [comment]
Worked with Komodo11.0.1 all day today with NO crashes.  Hooray!  Now if I could just get an alphabetical version of the simbol table in the right pane I would be a happy camper.

**Comment by Naatan** [comment]
Hit the cog > "Sort Alphabetically" :)

**Comment by Naatan** [comment]
@codermjb can you check if 11.0.1 solved your issue also?

**Comment by codermjb** [comment]
@Naatan 

I had about 12 tabs open today nothing big in size and I was writing a MySQL query in the code, where it just all of a sudden crashed and I lost my work.

**Comment by Naatan** [comment]
@codermjb Is this the first crash since last you responded?

**Comment by codermjb** [comment]
@Naatan Yes, but I haven't used it since. I am using it for a little bit for today (work has been a little different lately for me). I also will check to see if I got your latest update that isn't nightly build. I am assuming that right now the latest release was OTA and is not in the nightly builds repo? If so, I might need to check for updates via Help since it didn't prompt me.

**Comment by codermjb** [comment]
I got another crash today when I switched a tab. I only had about 4 or 5 tabs open and it crashed.

**Comment by codermjb** [comment]
Still continues to crash, loosing all my important code and having to reopen everything including the projects and finding the paths to all my individual several files I have. The autosave feature you have does not always work. And even when it prompts you, it doesn't open where you left off usually.

I feel it's something to do with how Komodo handles files. I think it's something with that syntax file checker that likes to scan thousands of files constantly. There isn't even a way to reduce this.

And now there is an update that if I press enter it adds random code like exp; and POST. Like seriously, I did not type that so don't randomly put code in at the end of my line `;`

**Comment by Naatan** [comment]
That sounds like https://github.com/Komodo/KomodoEdit/issues/3506

Have you experimented with turning off syntax checkers?
